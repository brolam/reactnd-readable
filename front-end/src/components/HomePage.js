import React from 'react'
import PropTypes from 'prop-types'
import PostCategoriesFilter from '../components/PostCategoriesFilter'
import SearchBar from '../components/SearchBar'
import PostList from '../components/PostList'
import PostModal from '../components/PostModal'
import WaitProcessModal from '../components/WaitProcessModal'

function HomePage({
  posts,
  categories,
  isNewPost = false,
  isShowWaitProcessModal = false,
  goPostNew = e => { },
  goHome = e => { },
  goHomeFilterByCategory = (categoryPath) => { },
  onSavePost = fieldsWasValidated => { },
  onSelectedPost = (post) => { },
  onVoteScorePost = (postId, option) => { },
  onChangeOrderPostsList = (order) => { } }) {
  return (
    <div className="app">
      <div className="main-page-header">
        <div className="main-page-header-title">
          <span />
          <h1>Readable</h1>
          <PostCategoriesFilter
            categories={categories}
            onSelected={e => { goHomeFilterByCategory(e.target.value) }} />
        </div>
      </div>
      <SearchBar placeholder="Search by title post" />
      <div className="main-page-content">
        <PostList
          posts={posts}
          onSelectedPost={onSelectedPost}
          onVoteScorePost={onVoteScorePost}
          onChangeOrderPostsList={onChangeOrderPostsList} />
      </div>
      {
        isNewPost && (
          <PostModal
            post={{}}
            categories={categories}
            onClickBackButton={goHome}
            onSavePost={onSavePost}
          />
        )
      }
      {
        isShowWaitProcessModal && (
          <WaitProcessModal
            message="Please wait while the information is updated."
          />
        )
      }
      <div className="flat-button" onClick={goPostNew}>
        <a className="add">Add Post</a>
      </div>
    </div>
  )
}

HomePage.propTypes = {
  posts: PropTypes.array.isRequired,
  categories: PropTypes.array.isRequired,
}

export default HomePage