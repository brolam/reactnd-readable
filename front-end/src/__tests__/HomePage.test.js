import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import HomePage from '../components/HomePage'

const categories = global.dataForTest.categories
const posts = global.dataForTest.posts

test('render without error', () => {
  ReactDOM.render(
    <HomePage posts={posts} categories={categories} />,
    document.createElement('div'));
})

test('last Snapshot', () => {
  const homePage = renderer.create(<HomePage posts={posts} categories={categories} />);
  let tree = homePage.toJSON();
  expect(tree).toMatchSnapshot();
})

test('last Snapshot - show new post form modal', () => {
  const homePage = renderer.create(<HomePage isNewPost={true} posts={posts} categories={categories} />);
  let tree = homePage.toJSON();
  expect(tree).toMatchSnapshot();
})

test('last Snapshot - show message wait process modal', () => {
  const homePage = renderer.create(
  <HomePage 
  posts={[]} 
  categories={[]}
  isShowWaitProcessModal={true}
   />);
  let tree = homePage.toJSON();
  expect(tree).toMatchSnapshot();
})

test('last Snapshot - show error message modal', () => {
  const homePage = renderer.create(
  <HomePage 
  posts={[]} 
  categories={[]}
  isShowErrorMessageModal={{message:"An error happened, please try again."}}
   />);
  let tree = homePage.toJSON();
  expect(tree).toMatchSnapshot();
})