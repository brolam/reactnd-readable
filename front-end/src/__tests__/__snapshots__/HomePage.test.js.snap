// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`last Snapshot - show error message modal 1`] = `
<div
  className="app"
>
  <div
    className="main-page-header"
  >
    <div
      className="main-page-header-title"
    >
      <span />
      <h1>
        Readable
      </h1>
      <select
        onChange={[Function]}
      >
        <option
          value="none"
        >
          All Categories
        </option>
      </select>
    </div>
  </div>
  <div
    className="search-bar"
  >
    <input
      placeholder="Search by title post"
      type="text"
    />
  </div>
  <div
    className="main-page-content"
  >
    <div
      className="posts"
    >
      <div
        className="posts-title"
      >
        <span>
          Posts
        </span>
        <div
          className="order-options"
        >
          <select>
            <option
              value="voteScore"
            >
              Vote Score
            </option>
            <option
              value="publishedDate"
            >
              Published date
            </option>
          </select>
        </div>
      </div>
      <div
        className="posts-list"
      />
    </div>
  </div>
  <div
    className="flat-button"
    onClick={[Function]}
  >
    <a
      className="add"
    >
      Add Post
    </a>
  </div>
</div>
`;

exports[`last Snapshot - show message wait process modal 1`] = `
<div
  className="app"
>
  <div
    className="main-page-header"
  >
    <div
      className="main-page-header-title"
    >
      <span />
      <h1>
        Readable
      </h1>
      <select
        onChange={[Function]}
      >
        <option
          value="none"
        >
          All Categories
        </option>
      </select>
    </div>
  </div>
  <div
    className="search-bar"
  >
    <input
      placeholder="Search by title post"
      type="text"
    />
  </div>
  <div
    className="main-page-content"
  >
    <div
      className="posts"
    >
      <div
        className="posts-title"
      >
        <span>
          Posts
        </span>
        <div
          className="order-options"
        >
          <select>
            <option
              value="voteScore"
            >
              Vote Score
            </option>
            <option
              value="publishedDate"
            >
              Published date
            </option>
          </select>
        </div>
      </div>
      <div
        className="posts-list"
      />
    </div>
  </div>
  <div
    className="modal-short modal-open"
  >
    <div
      className="modal-short-dialog wait-process-dialog"
    >
      <h1>
        Please wait while the information is updated.
      </h1>
    </div>
  </div>
  <div
    className="flat-button"
    onClick={[Function]}
  >
    <a
      className="add"
    >
      Add Post
    </a>
  </div>
</div>
`;

exports[`last Snapshot - show new post form modal 1`] = `
<div
  className="app"
>
  <div
    className="main-page-header"
  >
    <div
      className="main-page-header-title"
    >
      <span />
      <h1>
        Readable
      </h1>
      <select
        onChange={[Function]}
      >
        <option
          value="none"
        >
          All Categories
        </option>
        <option
          value="udacity"
        >
          Udacity
        </option>
        <option
          value="react"
        >
          React
        </option>
        <option
          value="redux"
        >
          Redux
        </option>
      </select>
    </div>
  </div>
  <div
    className="search-bar"
  >
    <input
      placeholder="Search by title post"
      type="text"
    />
  </div>
  <div
    className="main-page-content"
  >
    <div
      className="posts"
    >
      <div
        className="posts-title"
      >
        <span>
          Posts
        </span>
        <div
          className="order-options"
        >
          <select>
            <option
              value="voteScore"
            >
              Vote Score
            </option>
            <option
              value="publishedDate"
            >
              Published date
            </option>
          </select>
        </div>
      </div>
      <div
        className="posts-list"
      >
        <div
          className="post"
          onClick={[Function]}
        >
          <div
            className="post-header"
          >
            <div
              className="post-header-title"
            >
              <span
                className="category udacity"
              >
                Udacity
              </span>
               
              Udacity is the best place to learn technology.
            </div>
            <div
              className="post-header-published"
            >
              <span
                className="published"
              >
                a year ago
              </span>
            </div>
          </div>
          <div
            className="post-header-author"
          >
            by 
            thingtwo
          </div>
          <div
            className="post-body"
          >
            Everyone says so after all.
          </div>
          <div
            className="post-footer"
          >
            <button
              className="liked"
              onClick={[Function]}
            >
              Liked 
              <span>
                5
              </span>
            </button>
            <button
              className="not-liked"
              onClick={[Function]}
            >
              Not like
            </button>
          </div>
        </div>
        <div
          className="post"
          onClick={[Function]}
        >
          <div
            className="post-header"
          >
            <div
              className="post-header-title"
            >
              <span
                className="category react"
              >
                React
              </span>
               
              Udacity is the best place to learn React
            </div>
            <div
              className="post-header-published"
            >
              <span
                className="published"
              >
                a year ago
              </span>
            </div>
          </div>
          <div
            className="post-header-author"
          >
            by 
            thingtwo
          </div>
          <div
            className="post-body"
          >
            Everyone says so after all.
          </div>
          <div
            className="post-footer"
          >
            <button
              className="liked"
              onClick={[Function]}
            >
              Liked 
              <span>
                5
              </span>
            </button>
            <button
              className="not-liked"
              onClick={[Function]}
            >
              Not like
            </button>
          </div>
        </div>
        <div
          className="post"
          onClick={[Function]}
        >
          <div
            className="post-header"
          >
            <div
              className="post-header-title"
            >
              <span
                className="category redux"
              >
                Redux
              </span>
               
              Udacity is the best place to learn Redux
            </div>
            <div
              className="post-header-published"
            >
              <span
                className="published"
              >
                a year ago
              </span>
            </div>
          </div>
          <div
            className="post-header-author"
          >
            by 
            thingtwo
          </div>
          <div
            className="post-body"
          >
            Everyone says so after all.
          </div>
          <div
            className="post-footer"
          >
            <button
              className="liked"
              onClick={[Function]}
            >
              Liked 
              <span>
                5
              </span>
            </button>
            <button
              className="not-liked"
              onClick={[Function]}
            >
              Not like
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="modal modal-open"
    id="postModal"
  >
    <div
      className="modal-dialog"
    >
      <div
        className="modal-heard modal-post"
      >
        <span
          onClick={[Function]}
        />
        <input
          autoFocus={true}
          defaultValue={undefined}
          maxLength="80"
          minLength="10"
          placeholder="Title post"
          required={true}
          type="text"
        />
      </div>
      <div
        className="modal-content modal-post"
      >
        <textarea
          defaultValue={undefined}
          maxLength="500"
          minLength="2"
          placeholder="Body post"
          required={true}
        />
        <div
          className="modal-footer"
        >
          <select
            onChange={[Function]}
          >
            <option
              value="none"
            >
              Save as?
            </option>
            <option
              value="udacity"
            >
              Udacity
            </option>
            <option
              value="react"
            >
              React
            </option>
            <option
              value="redux"
            >
              Redux
            </option>
          </select>
        </div>
      </div>
    </div>
  </div>
  <div
    className="flat-button"
    onClick={[Function]}
  >
    <a
      className="add"
    >
      Add Post
    </a>
  </div>
</div>
`;

exports[`last Snapshot 1`] = `
<div
  className="app"
>
  <div
    className="main-page-header"
  >
    <div
      className="main-page-header-title"
    >
      <span />
      <h1>
        Readable
      </h1>
      <select
        onChange={[Function]}
      >
        <option
          value="none"
        >
          All Categories
        </option>
        <option
          value="udacity"
        >
          Udacity
        </option>
        <option
          value="react"
        >
          React
        </option>
        <option
          value="redux"
        >
          Redux
        </option>
      </select>
    </div>
  </div>
  <div
    className="search-bar"
  >
    <input
      placeholder="Search by title post"
      type="text"
    />
  </div>
  <div
    className="main-page-content"
  >
    <div
      className="posts"
    >
      <div
        className="posts-title"
      >
        <span>
          Posts
        </span>
        <div
          className="order-options"
        >
          <select>
            <option
              value="voteScore"
            >
              Vote Score
            </option>
            <option
              value="publishedDate"
            >
              Published date
            </option>
          </select>
        </div>
      </div>
      <div
        className="posts-list"
      >
        <div
          className="post"
          onClick={[Function]}
        >
          <div
            className="post-header"
          >
            <div
              className="post-header-title"
            >
              <span
                className="category udacity"
              >
                Udacity
              </span>
               
              Udacity is the best place to learn technology.
            </div>
            <div
              className="post-header-published"
            >
              <span
                className="published"
              >
                a year ago
              </span>
            </div>
          </div>
          <div
            className="post-header-author"
          >
            by 
            thingtwo
          </div>
          <div
            className="post-body"
          >
            Everyone says so after all.
          </div>
          <div
            className="post-footer"
          >
            <button
              className="liked"
              onClick={[Function]}
            >
              Liked 
              <span>
                5
              </span>
            </button>
            <button
              className="not-liked"
              onClick={[Function]}
            >
              Not like
            </button>
          </div>
        </div>
        <div
          className="post"
          onClick={[Function]}
        >
          <div
            className="post-header"
          >
            <div
              className="post-header-title"
            >
              <span
                className="category react"
              >
                React
              </span>
               
              Udacity is the best place to learn React
            </div>
            <div
              className="post-header-published"
            >
              <span
                className="published"
              >
                a year ago
              </span>
            </div>
          </div>
          <div
            className="post-header-author"
          >
            by 
            thingtwo
          </div>
          <div
            className="post-body"
          >
            Everyone says so after all.
          </div>
          <div
            className="post-footer"
          >
            <button
              className="liked"
              onClick={[Function]}
            >
              Liked 
              <span>
                5
              </span>
            </button>
            <button
              className="not-liked"
              onClick={[Function]}
            >
              Not like
            </button>
          </div>
        </div>
        <div
          className="post"
          onClick={[Function]}
        >
          <div
            className="post-header"
          >
            <div
              className="post-header-title"
            >
              <span
                className="category redux"
              >
                Redux
              </span>
               
              Udacity is the best place to learn Redux
            </div>
            <div
              className="post-header-published"
            >
              <span
                className="published"
              >
                a year ago
              </span>
            </div>
          </div>
          <div
            className="post-header-author"
          >
            by 
            thingtwo
          </div>
          <div
            className="post-body"
          >
            Everyone says so after all.
          </div>
          <div
            className="post-footer"
          >
            <button
              className="liked"
              onClick={[Function]}
            >
              Liked 
              <span>
                5
              </span>
            </button>
            <button
              className="not-liked"
              onClick={[Function]}
            >
              Not like
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="flat-button"
    onClick={[Function]}
  >
    <a
      className="add"
    >
      Add Post
    </a>
  </div>
</div>
`;
